src/cpp/proxy/sockjsmanager.cpp

```diff
@@ -617,10 +621,9 @@ private slots:
 		}
 	}
 
-	void req_error()
+	void req_error(ZhttpRequest *req)
 	{
-		ZhttpRequest *req = (ZhttpRequest *)sender();
-
+		reqConnectionMap.erase(req);
```

Comment:
This should probably be moved down to only be called when doing `delete req`.

Reply:
the reason I added map erase here is that I saw the logic here:
```
		if(discardedRequests.contains(req))
		{
			discardedRequests.remove(req);
			delete req;
			return;
		}
```
and I thought map needs an erase regardless of req existing in discardedRequests or not ... later on in the func we have 
```		Session *s = sessionsByRequest.value(req);
		assert(s);

		if(s->pending)
			s->req = 0;
		else
			removeSession(s);
```

Reply:
the code is really confusing but I think disassociating a request from a session, and even removing the session, shouldn't destroy the request. at least this seems to be something that happens in other places. such a request would presumably be destroyed later on somehow

Reply:
notably, if the request was supposed to remain alive with its signals disconnected, there would have been a qt-style disconnect call here

Reply:
ah, destroying the session does destroy the request, so that case will need to be handled. I suggest disconnecting the signals from `~Session` via `owner` if possible, else from `removeSession`.

Reply:
and only do this if the `req` member is non-null

Reply:
got it so I added `reqConnectionMap.erase(req);` right next to where ever there is a `delete req` and also to `removeSession`